### PR TITLE
runtime/common: Implement Add, AddAssign, Sub, SubAssign for Quantity

### PR DESCRIPTION
**Pull request by kostko**

### CODE REVIEW

**Comment by Yawning** [comment]
Should this have any specific (read: non-standard) underflow behavior?

File: runtime/src/common/quantity.rs [hunk 1]
```diff
@@ -15,6 +18,36 @@ impl From<u64> for Quantity {
     }
 }
 
+impl<'a> Add<&'a Quantity> for Quantity {
+    type Output = Quantity;
+
+    fn add(mut self, other: &Quantity) -> Quantity {
```
Comment (ptrus) [inline runtime/src/common/quantity.rs:24]: Maybe i'm missing something, but shouldn't `Add`  not update `self`?

Comment (kostko) [reply runtime/src/common/quantity.rs:24]: So it is not `&mut self` but `mut self` (as required by the `Add` trait) which means that self is moved into the method. This means that there are no other references to it anywhere so it is safe to reuse it.

**Comment by kostko** [comment]
> Should this have any specific (read: non-standard) underflow behavior?

Yeah it would be safer if we only provided `checked_sub`, will change it.

**Comment by kostko** [comment]
Done, the use is now forced to be as follows:
```rust
        value = value
            .checked_sub(&amount.amount())
            .ok_or(Error::InsufficientBalance)?;
```
or
```rust
        total_supply = total_supply
            .checked_sub(&amount.amount())
            .expect("target account had enough balance so total supply should not underflow");
```

File: runtime/src/common/quantity.rs (hunk 1 above)
Comment (ptrus) [reply runtime/src/common/quantity.rs]: Ah i see. Thanks :+1:

**Review by ptrus — APPROVED** [review]

**Pull request merged**